src/libs/antares/study/parts/renewable/cluster.cpp

```diff
@@ -233,6 +227,48 @@ const char* Data::RenewableCluster::GroupName(enum RenewableGroup grp)
     return "";
 }
 
+bool Data::RenewableCluster::setTimeSeriesModeFromString(const YString& value)
+{
+    if (value == "power-generation")
+    {
+        tsMode = powerGeneration;
+        return true;
+    }
+    if (value == "production-factor")
+    {
+        tsMode = productionFactor;
+        return true;
+    }
+    return false;
+}
+
+const YString& Data::RenewableCluster::getTimeSeriesModeAsString() const
+{
+    switch (tsMode)
+    {
+    case powerGeneration:
+        return "power-generation";
+    case productionFactor:
+        return "production-factor";
+    }
+    return "unknown";
+}
+
+double RenewableCluster::valueAtTimeStep(uint timeSeriesIndex, uint timeStepIndex) const
+{
+    assert(timeStepIndex < series->series.height);
+    assert(timeSeriesIndex < series->series.width);
+    const double tsValue = series->series[timeSeriesIndex][timeStepIndex];
+    switch (tsMode)
+    {
+    case powerGeneration:
+        return unitCount * tsValue;
```

Comment:
In case time series represent power generation, **tsValue** alone is the power generation : we should not have to multiply **tsValue** by **unitCount**  to get the cluster's generation